applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp

```diff
@@ -476,6 +476,11 @@ void FluidElement<TElementData>::GetValueOnIntegrationPoints(
 
         VorticityUtilities<Dim>::CalculateVorticityMagnitude(this->GetGeometry(),shape_function_gradients,rValues);
     }
+    else if (rVariable == UPDATE_STATISTICS)
+    {
+        KRATOS_DEBUG_ERROR_IF(!rCurrentProcessInfo.Has(STATISTICS_CONTAINER)) << "Trying to compute turbulent statistics, but ProcessInfo does not have STATISTICS_CONTAINER defined." << std::endl;
+        rCurrentProcessInfo.GetValue(STATISTICS_CONTAINER)->UpdateStatistics(this);
```

Comment:
is UpdateStatistics threadsafe?

Reply:
The only risk for thread safety I could think of would be the `mUpdateBuffer` you noticed in the other comments, so it should be taken care of.